bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/ExperienceFragmentImpl.java

```diff
@@ -0,0 +1,221 @@
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.internal.models.v1;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import javax.jcr.RangeIterator;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.models.annotations.Exporter;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.injectorspecific.*;
+import org.jetbrains.annotations.NotNull;
+
+import com.adobe.cq.export.json.ComponentExporter;
+import com.adobe.cq.export.json.ExporterConstants;
+import com.adobe.cq.wcm.core.components.models.ExperienceFragment;
+import com.day.cq.wcm.api.LanguageManager;
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.Template;
+import com.day.cq.wcm.api.WCMException;
+import com.day.cq.wcm.msm.api.LiveCopy;
+import com.day.cq.wcm.msm.api.LiveRelationship;
+import com.day.cq.wcm.msm.api.LiveRelationshipManager;
+import com.day.text.Text;
+
+@Model(adaptables = SlingHttpServletRequest.class,
+    adapters = {ExperienceFragment.class, ComponentExporter.class },
+    resourceType = {ExperienceFragmentImpl.RESOURCE_TYPE_V1 })
+@Exporter(name = ExporterConstants.SLING_MODEL_EXPORTER_NAME,
+    extensions = ExporterConstants.SLING_MODEL_EXTENSION)
+public class ExperienceFragmentImpl implements ExperienceFragment {
+
+    public static final String RESOURCE_TYPE_V1 = "core/wcm/components/experiencefragment/v1/experiencefragment";
+
+    private static final String PATH_DELIMITER = "/";
+    private static final char PATH_DELIMITER_CHAR = '/';
+    private static final String CONTENT_ROOT = "/content";
+    private static final String EXPERIENCE_FRAGMENTS_ROOT = "/content/experience-fragments";
+    private static final String JCR_CONTENT_ROOT = "/jcr:content";
+
+    @Self
+    private SlingHttpServletRequest request;
+
+    @Inject
+    protected Resource resource;
+
+    @SlingObject
+    private ResourceResolver resolver;
+
+    @ScriptVariable
+    private Page currentPage;
+
+    @ValueMapValue(name = ExperienceFragment.PN_FRAGMENT_PATH, injectionStrategy = InjectionStrategy.OPTIONAL)
+    private String fragmentPath;
+
+    @OSGiService
+    private LanguageManager languageManager;
+
+    @OSGiService
+    private LiveRelationshipManager relationshipManager;
+
+    private String localizedFragmentPath;
+
+    @PostConstruct
+    protected void initModel() {
+        if (inTemplate()) {
```

Comment:
should we consider following cases:
an XF is placed on a page. Currently the view layer refers to `localizedFragmentPath` only. The `localizedFragmentPath` is calculated only in particular cases when an XF is placed on a template so in result nothing will be displayed
please correct me if I am wrong